### PR TITLE
Added the SecurityContext for Driver

### PR DESCRIPTION
**Pull request by mskanth972**
**Is this a bug fix or adding new feature?**

**What is this PR about? / Why do we need it?**
This PR adds the Security Context for the EFS CSI driver.

**What testing is done?** 
Deployed the CSI driver with the latest yaml which has the changes and working fine.

### CODE REVIEW

File: charts/aws-efs-csi-driver/values.yaml [hunk 1]
```diff
@@ -136,6 +153,13 @@ node:
     ## Enable if EKS IAM for SA is used
     #  eks.amazonaws.com/role-arn: arn:aws:iam::111122223333:role/efs-csi-role
   healthPort: 9809
+  # securityContext on the node pod
+  securityContext:
+    # The node pod must be run as root to bind to the registration/driver sockets
+    runAsNonRoot: false
```
Comment (Ashley-wenyizha) [inline charts/aws-efs-csi-driver/values.yaml:159]: Bit confused here

From the comment it's saying that `The node pod must be run as root to bind to the registration/driver sockets`

but then why we set `runAsNonRoot: false` ?

File: charts/aws-efs-csi-driver/values.yaml [hunk 2]
```diff
@@ -80,6 +89,14 @@ controller:
     #  eks.amazonaws.com/role-arn: arn:aws:iam::111122223333:role/efs-csi-role
   healthPort: 9909
   regionalStsEndpoints: false
+  # securityContext on the controller pod
+  securityContext:
+    runAsNonRoot: false
+    runAsUser: 0
```
[the remark this answers is not in the document]
Comment (Ashley-wenyizha) [reply charts/aws-efs-csi-driver/values.yaml:95]: Same to other '0's we set below and in other yaml files

File: charts/aws-efs-csi-driver/values.yaml (hunk 1 above)
Comment (mskanth972) [reply charts/aws-efs-csi-driver/values.yaml:159]: `runAsNonRoot: false` which means `root: true`. We are explicitly declaring important security configurations.

File: charts/aws-efs-csi-driver/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mskanth972) [reply charts/aws-efs-csi-driver/values.yaml:95]: Only efs-plugin needs the root user to mount the efs storage at nodes root level folder. Other containers are running as non root.

[the remark this answers is not in the document]
Comment (mauriciopoppe) [reply charts/aws-efs-csi-driver/values.yaml:95]: I believe the Pod & container `securityContext` config is merged but I don't see the override in the containers. As it is right now are the containers running as root now?

**Comment by TobiasRD** [comment]
Hey @mskanth972 when updating the helm chart to 2.4.8, which includes this change, I get trouble with the readOnlyRootFilesystem=true for the csi-driber-registrar. 
I get the following error: 
`"Failed to create registration probe file" err="mkdir /var/lib/kubelet: read-only file system" registrationProbePath="/var/lib/kubelet/plugins/efs.csi.aws.com/registration"`

Do you have any idea? (beside setting it to false)

**Comment by mauriciopoppe** [comment]
@TobiasRD this is a known issue in node-driver-registrar which will be fixed soon in https://github.com/kubernetes-csi/node-driver-registrar/pull/316. The workaround is to not use readOnlyRootFilesystem for now but once there's a new node-driver-registrar release with the above fix then you'd be able to use `readOnlyRootFilesystem: true`.

**Comment by mskanth972** [comment]
> @TobiasRD this is a known issue in node-driver-registrar which will be fixed soon in [kubernetes-csi/node-driver-registrar#316](https://github.com/kubernetes-csi/node-driver-registrar/pull/316). The workaround is to not use readOnlyRootFilesystem for now but once there's a new node-driver-registrar release with the above fix then you'd be able to use `readOnlyRootFilesystem: true`.

Thanks for pointing the known issue @mauriciopoppe

**Comment by mauriciopoppe** [comment]
@TobiasRD it doesn't affect the functionality, if you see the error consider it a false negative error :(.

**Pull request merged**